src/utils/fontface.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Guess the font's CSS "format" by analysing the asset path.
+ */
+function getFontCSSFormat(path, errorOnStyle) {
+  let pos = path.lastIndexOf(`.`);
+  let ext = (path.substring(pos + 1) || ``).toLowerCase();
+
+  let format = {
+    ttf: `truetype`,
+    otf: `opentype`,
+    woff: `woff`,
+    woff2: `woff2`,
+  }[ext];
+
+  if (format) return format;
+
+  let msg = {
+    eot: `The .eot format is not supported: it died in January 12, 2016, when Microsoft retired all versions of IE that didn't already support WOFF.`,
+    svg: `The .svg format is not supported: SVG fonts (not to be confused with OpenType with embedded SVG) were so bad we took the entire fonts chapter out of the SVG specification again.`,
+    fon: `The .fon format is not supported: this is an ancient Windows bitmap font format.`,
+    ttc: `Based on the current CSS specification, font collections are not (yet?) supported.`,
+  }[ext];
+
+  if (!msg) msg = `${path} is not a known webfont format.`;
+
+  if (errorOnStyle) {
+    // hard stop if the user wants stylesheet errors to count as true errors,
+    throw new Error(msg);
+  } else {
+    // otherwise, only leave a warning in the output log.
+    console.warn(`Could not load font: ${msg}`);
+  }
```

Comment:
Let the user decide whether this should be an error, or just a warning (because even in browser land they might not care if the CSS part fails, as long as the font part succeeds)

Reply:
👍